### ISSUE TITLE
Bug report - provider consistently crashes when trying to create a vault_mfa_duo object in a module

### ISSUE THREAD

**Issue by benjamin-issa** [hashicorp/terraform-provider-vault#1527]
### Terraform Version
```
2022-07-08T19:12:11.271Z [INFO]  Terraform version: 1.1.7
2022-07-08T19:12:11.271Z [INFO]  Go runtime version: go1.17.2
2022-07-08T19:12:11.271Z [INFO]  CLI args: []string{"terraform", "-v"}
2022-07-08T19:12:11.271Z [TRACE] Stdout is a terminal of width 218
2022-07-08T19:12:11.271Z [TRACE] Stderr is a terminal of width 218
2022-07-08T19:12:11.271Z [TRACE] Stdin is a terminal
2022-07-08T19:12:11.271Z [DEBUG] Attempting to open CLI config file: /root/.terraformrc
2022-07-08T19:12:11.271Z [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2022-07-08T19:12:11.272Z [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2022-07-08T19:12:11.273Z [DEBUG] ignoring non-existing provider search directory /root/.terraform.d/plugins
2022-07-08T19:12:11.273Z [DEBUG] ignoring non-existing provider search directory /root/.local/share/terraform/plugins
2022-07-08T19:12:11.273Z [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
2022-07-08T19:12:11.273Z [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
2022-07-08T19:12:11.273Z [INFO]  CLI command args: []string{"version", "-v"}
Terraform v1.1.7
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v3.75.2
+ provider registry.terraform.io/hashicorp/tls v3.4.0
+ provider registry.terraform.io/hashicorp/vault v3.7.
```

### Affected Resource(s)
- vault_mfa_duo.duo_mfa

### Terraform Configuration Files
Attached (see below), encrypted using [Hashicorp Security's GPG key](https://www.hashicorp.com/security) (please let me know if this was the wrong key to use).

https://github.com/benjamin-issa/terraform-vault-provider-bug

### Debug Output
See attached, above.

### Panic Output
Excerpt below; full log linked in attachment above.
```
Stack trace from the terraform-provider-vault_v3.7.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xefdbda]

goroutine 44 [running]:
github.com/hashicorp/terraform-provider-vault/vault.mfaDuoRead(0x68, {0x10a9ee0, 0xc000a60ae0})
        github.com/hashicorp/terraform-provider-vault/vault/resource_mfa_duo.go:122 +0x1fa
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x14a0030, {0x14a0030, 0xc00065bdd0}, 0xd, {0x10a9ee0, 0xc000a60ae0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000223500, {0x14a0030, 0xc00065bdd0}, 0xc00066a820, {0x10a9ee0, 0xc000a60ae0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:1015 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00000d2d8, {0x149ff88, 0xc0007f6f40}, 0xc0007f6fc0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/grpc_provider.go:613 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0000d9680, {0x14a0030, 0xc00065b620}, 0xc00013b980)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/tf5server/server.go:746 +0x48c
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x11bbce0, 0xc0000d9680}, {0x14a0030, 0xc00065b620}, 0xc00013b920, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002148c0, {0x14b7388, 0xc0002a61a0}, 0xc0005478c0, 0xc0006637a0, 0x1cea570, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0002148c0, {0x14b7388, 0xc0002a61a0}, 0xc0005478c0, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1619 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.45.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.45.0/server.go:919 +0x294

Error: The terraform-provider-vault_v3.7.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior
The object should have been successfully imported (or the terraform apply should either be successful or report a graceful error).

### Actual Behavior
What actually happened?
The vault plugin crahsed

### Steps to Reproduce
Once the terrform state gets in this state, most actions appear to cause the plugin to crash, including:
1. `terraform apply`
2. `terraform import module.lab_vault_configuration.vault_mfa_duo.duo_mfa duo`

### Important Factoids
Running in a development environment container.

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? 
No.

**Comment by trodemaster** [comment]
I'm seeing a similar crash with v3.8.2 of the plugin. 

```
Stack trace from the terraform-provider-vault_v3.8.2_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1ad4da1]

goroutine 94 [running]:
github.com/hashicorp/terraform-provider-vault/vault.identityGroupAliasUpdate(0xc000968400, {0x1cba520, 0xc000a0e120})
        github.com/hashicorp/terraform-provider-vault/vault/resource_identity_group_alias.go:93 +0x301
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x20e8710, {0x20e8710, 0xc00027d830}, 0xd, {0x1cba520, 0xc000a0e120})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:729 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000593960, {0x20e8710, 0xc00027d830}, 0xc000984000, 0xc000968280, {0x1cba520, 0xc000a0e120})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:847 +0x9e5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0005654a0, {0x20e8668, 0xc00094e300}, 0xc0002b2be0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0007645a0, {0x20e8710, 0xc00027d230}, 0xc0003683f0)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/tf5server/server.go:812 +0x56b
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1dcd460, 0xc0007645a0}, {0x20e8710, 0xc00027d230}, 0xc00027af60, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000ea8c0, {0x20ffc48, 0xc000290680}, 0xc000ac9680, 0xc00058d1d0, 0x2940760, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0000ea8c0, {0x20ffc48, 0xc000290680}, 0xc000ac9680, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1619 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.45.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.45.0/server.go:919 +0x294

Error: The terraform-provider-vault_v3.8.2_x5 plugin crashed!
```

**Comment by vinay-gopalan** [comment]
Hi @benjamin-issa, thanks for opening this issue! There was a panic in the Vault MFA Duo resource read that was fixed and released as part of [v3.11.0](https://github.com/hashicorp/terraform-provider-vault/releases/tag/v3.11.0). Going to close this issue as completed, but please feel free to open another one if you face any other problems with the resource. Thanks!
